stl/inc/condition_variable

```diff
@@ -70,21 +70,7 @@ public:
 
     template <class _Lock, class _Rep, class _Period>
     cv_status wait_for(_Lock& _Lck, const chrono::duration<_Rep, _Period>& _Rel_time) {
-        // wait for duration
-        if (_Rel_time <= chrono::duration<_Rep, _Period>::zero()) {
-            return cv_status::timeout;
-        }
-
-        // TRANSITION, ABI: The standard says that we should use a steady clock,
-        // but unfortunately our ABI speaks struct xtime, which is relative to the system clock.
-        _CSTD xtime _Tgt;
-        const bool _Clamped     = _To_xtime_10_day_clamped(_Tgt, _Rel_time);
-        const cv_status _Result = wait_until(_Lck, &_Tgt);
-        if (_Clamped) {
-            return cv_status::no_timeout;
-        }
-
-        return _Result;
+        return wait_until(_Lck, chrono::steady_clock::now() + _Rel_time);

```

Comment:
It's unfortunate that we call `steady_clock::now()` here and immediately again on line 85. Could we avoid this with some clever refactoring?

Reply:
If I substitute in `wait_until`'s fixed definition according to your suggestion, and eliminate `now - now` (which we can assume to be zero given infinitely fast execution), I just get the original definition of this function except missing the early return for `if (_Rel_time <= chrono::duration<_Rep, _Period>::zero())`. So should that just be the change here?

Reply:
Yes, I think that sounds great.

Reply:
@CaseyCarter , @StephanTLavavej  - Original bug was in the early return for `_Rel_time <= chrono::duration<_Rep, _Period>::zero()` as the code was skipping unlock and lock. I think the code is easier to understand and maintain if we don't try to optimize away those cases. Thoughts?

Reply:
It's true that the current state of the PR centralizes more of the logic, which is usually good for understanding and maintenance as you mentioned. However, translating from the relative time domain to absolute time, and back to relative, adds an additional point of confusion for understanding, adds performance cost, and affects the behavior slightly (as the elapsed time between the two `now()` calls affects reasoning about the code).

I believe that going back to the original implementation here (minus the early, incorrect return) would be more desirable - it would be more lines of code, but without the repeated `now()` calls with their performance and conceptual complexity cost. (If we were merely recomputing some pure value, I might feel differently, but `now()` is an OS call.)

Reply:
Sounds good. Let me send a new iteration. Most likely later in the week or weekend.